mscore/preferenceslistwidget.h

```diff
@@ -30,15 +30,17 @@ namespace Ms {
 //---------------------------------------------------------
 //   PreferenceItem
 //---------------------------------------------------------
-class PreferenceItem : public QTreeWidgetItem, public QObject {
+class PreferenceItem : public QObject, public QTreeWidgetItem {
+
+      Q_OBJECT
 
       QString _name;
 
     protected:
       void save(QVariant value);
 
     public:
-      PreferenceItem();
+      PreferenceItem() {};
       PreferenceItem(QString name);
```

Comment:
Why not make `_applyFunction` and `_updateFunction` members of `PreferenceItem`, so that this constructor can be changed to:

    PreferenceItem(const QString& name, std::function<void()> applyFunc = nullptr, std::function<void()> updateFunc = nullptr);

It would save a lot of duplicated boilerplate code in the subclasses.

(Side note: `QString` objects should be passed by `const &` wherever possible.)

Then you could also add the following member functions to `PreferenceItem`, again saving some boilerplate:

    bool PreferenceItem::callApplyFunction()
          {
          if (_applyFunction)
                _applyFunction();
    
          return _applyFunction;
          }
    
    bool PreferenceItem::callUpdateFunction()
          {
          if (_updateFunction)
                _updateFunction();
    
          return _updateFunction;
          }

Reply:
I don't get how the first suggested change helps with the boilerplate code. Could you, please, elaborate?